File: TimeFlies/vc/home/TimeSelectController.m
//
//  TimeSelectController.m
//  testTabbarController
//
//  Created by 林伟池 on 15/10/26.
//  Copyright © 2015年 林伟池. All rights reserved.
//

#import "TimeSelectController.h"
#import "HomeModel.h"

@interface TimeSelectController ()

@end

@implementation TimeSelectController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    [self.myDatePicker setMaximumDate:[[NSDate alloc] initWithTimeIntervalSinceNow:0]];
    [self.myDatePicker setDate:[[HomeModel instance] getBirthDate]];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

#pragma mark - view init

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
//    NSLog(@"appear");
}

- (void)viewWillDisappear:(BOOL)animated
{
    [super viewWillDisappear:animated];
//    NSLog(@"dis");
}

- (void)viewDidDisappear:(BOOL)animated
{
    [super viewDidDisappear:animated];
//    NSLog(@"dis %@", self.view.superview);
}
#pragma mark - ui

- (IBAction)onClose:(id)sender
{
    [self dismissViewControllerAnimated:YES completion:nil];
//    [self.lyDelegate dismissViewController:self];
}

- (IBAction)onSave:(id)sender
{
    [[HomeModel instance] setBirthDate:[self.myDatePicker date]];
    [self onClose:sender];
}

- (IBAction)onValueChange:(id)sender
{
    NSLog(@"%@", [self.myDatePicker date]);
}
#pragma mark - delegate

#pragma mark - notify


@end

File: TimeFlies/vc/home/TimeSelectController.swift
//
//  SwiftTimeSelectController.swift
//  TimeFlies
//
//  Created by 林伟池 on 16/4/8.
//  Copyright © 2016年 林伟池. All rights reserved.
//

import Foundation

/**
 
*/
protocol SwiftModalViewControllerDelegate {
    func dismissViewController(_ controller: SwiftTimeSelectController)
}

class SwiftTimeSelectController: UIViewController {
    
    //MARK: property
    @IBOutlet var myDatePicker:UIDatePicker?
    var lyDelegate:SwiftModalViewControllerDelegate?
    
    //MARK: init
    override func viewDidLoad() {
        super.viewDidLoad()
        
        myDatePicker?.maximumDate = Date(timeIntervalSinceNow: 0)
        myDatePicker?.date = HomeModel.instance().getBirthDate()
        

        //TEST weak strong dance
        weak var weakSelf = self
        NotificationCenter.default.addObserver(forName: NSNotification.Name(rawValue: "EssayModelChange"), object: nil, queue: nil) { (note) -> Void in
            let strongSelf = weakSelf
            print("EssayModalChange is listen \(strongSelf)")

        }
    }
    
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    override func viewDidDisappear(_ animated: Bool) {
        super.viewDidDisappear(animated)
    }
    
    deinit {
        print("dealloc \(self)")
    }
    
    //MARK: ui
    @IBAction func onClose(_ sender: NSObject) {
        dismiss(animated: true, completion: nil)
    }
    
    @IBAction func onSave(_ sender: NSObject) {
        HomeModel.instance().setBirthDate(myDatePicker?.date)
        onClose(sender)
    }
    
    @IBAction func onValueChange(_ sender: NSObject) {
        print("date with \(myDatePicker?.date)")
    }
    
    
    
}
